### ISSUE TITLE
2.8.2 cannot connect to PBIX using Direct Query against Oracle database

### ISSUE THREAD

**Issue by SamTrexler** [DaxStudio/DaxStudio#164]
I have downloaded and installed the latest release of DAX Studio, 2.8.2, in order to test and debug models using aggregations. 

I can connect to pbix files that import data from an Oracle database, but not those that use direct query. The database connection uses the recommended ODBC driver, but I have to specify the connection as "(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=x.x.x.x)(PORT=xxxx))(CONNECT_DATA=(SID=sidname)))"

Is there a way to connect to these models so I can test and get server timings, etc.?

**Comment by dgosbell** [comment]
Can you download and test the 2.8.3 version that was just released? This fixes an issue around connecting to PBIX models that use Direct Query connections. If you are seeing and error saying to "check that the server is running" then this should fix your issue

**Comment by SamTrexler** [comment]
That's exactly the error I am getting. I'll download the new version and give it a try. Thanks.

**Comment by SamTrexler** [comment]
I have tried 2.8.3 against two or three Direct Query reports, and it does not solve the problem. I still get the error "Error Connecting: A connection cannot be made. Ensure that the server is running."
![DAX Studio Connection Error](https://user-images.githubusercontent.com/48493709/54303325-f8de6b80-457f-11e9-9610-9bae4fe344b6.jpg)

**Comment by dgosbell** [comment]
Hmm, maybe this is a new issue. Can you enable Debug logging using the steps [here](https://daxstudio.org/documentation/troubleshooting/logging-settings/) then try and connect to a DQ PBIX model then close DAX Studio and attach the log file back to this issue.

Then can you also run the following PowerShell (either in a PowerShell window or in PowerShell ISE)
and paste the output back here or attach it in a txt file?

```
$process = "MSMDSRV"
get-process -name $process | % {get-nettcpconnection -OwningProcess $_.id} | select-object @{Name="Process"; Expression = {$Process}},LocalPort, RemotePort, OwningProcess
$process = "PBIDesktop"
get-process -name $process | % {get-nettcpconnection -OwningProcess $_.id} | select-object @{Name="Process"; Expression = {$Process}},LocalPort, RemotePort, OwningProcess
```

This will allow me to see what ports PowerBI had open and what ports DAX Studio detected

**Comment by SamTrexler** [comment]
Here's what I got.
[DaxStudio-20190313.log](https://github.com/DaxStudio/DaxStudio/files/2964131/DaxStudio-20190313.log)
[DAX Studio Powershell Debugging.txt](https://github.com/DaxStudio/DaxStudio/files/2964133/DAX.Studio.Powershell.Debugging.txt)

**Comment by dgosbell** [comment]
OK, so it looks like the detection code is finding one of the "outgoing" ports to Oracle instead of the correct "incoming" port. I have a sneaking suspicion that the Oracle driver might behave differently than SQL Server. But I don't have an instance of Oracle that I can double check this against. 

When you get a chance can you do the same test again but using the PowerShell below that adds the state of the ports:
```
$process = "MSMDSRV"
get-process -name $process | % {get-nettcpconnection -OwningProcess $_.id} | select-object @{Name="Process"; Expression = {$Process}},LocalPort, RemotePort, OwningProcess, State | ft
$process = "PBIDesktop"
get-process -name $process | % {get-nettcpconnection -OwningProcess $_.id} | select-object @{Name="Process"; Expression = {$Process}},LocalPort, RemotePort, OwningProcess, State | ft
```
If my theory is correct I might have to change the way I detect the port for PowerBI (there is a slightly more involved method which should work around this issue)

**Comment by SamTrexler** [comment]
Okay, here's what happened this time. I hope I captured everything, the log file looks pretty small.

[DaxStudio-20190314.log](https://github.com/DaxStudio/DaxStudio/files/2969005/DaxStudio-20190314.log)
[DAX Studio Powershell Debugging 2019-03-14.txt](https://github.com/DaxStudio/DaxStudio/files/2969006/DAX.Studio.Powershell.Debugging.2019-03-14.txt)

**Comment by dgosbell** [comment]
Thanks @SamTrexler the logs look fine although in this instance it looks like DAX Studio found the same ports that Power BI Desktop was using. Did you see the same connection errors when you captured the second set of logs or is this error intermittent? I'm not seeing the issue in the logs that I was expecting so I'm still not sure why the fix in 2.8.3 is not working consistently here. I've had a few other people report issues connecting to Direct Query models and 2.8.3 seems to have fixed this for them although I think they were using either SQL Server or AzureSQL

**Comment by dgosbell** [comment]
Closing due to lack of response - feel free to reopen or raise a new issue if you have more information to add
